src/Azure.Functions.Cli/Common/KeyVaultReferencesManager.cs

```diff
@@ -45,19 +48,34 @@ private string GetSecretValue(string value)
 
         private ParseSecretResult ParseSecret(string value)
         {
-            try
+            // If the value is null, then we return nothing, as the subsequent call to
+            // UpdateEnvironmentVariables(settings) will log to the user that the setting
+            // is skipped. We check here, because Regex.Match throws when supplied with a
+            // null value.
+            if (value == null)
             {
-                return ParsePrimaryRegexSecret(value) ?? ParseSecondaryRegexSecret(value);
+                return null;
```

Comment:
I like the extra checks, but I think we still need an overarching try/catch somewhere so that we can be 100% sure this won't block `func start`

I would suggest putting the try/catch either in `GetSecretValue` or within the for loop of `ResolveKeyVaultReferences` so that
- it covers both `ParseSecret` and `client.GetSecret`
- if one secret value is bad, we'll still try the other ones